mocha_spec/integration_helpers/index.js

```diff
@@ -41,37 +41,28 @@ class AmqpHelper extends EventEmitter {
         env.ELASTICIO_TIMEOUT = 3000;
     }
 
-    // optional callback `done` is used in order to pass exceptions (e.g. from assertions in tests) to mocha callback
-    on(event, handler, done = undefined) {
-        if (!done) {
-            return super.on(event, handler);
-        }
-
-        return super.on(event, (...args) => {
-            try {
-                handler(...args);
-                done();
-            } catch (e) {
-                done(e);
-            }
-        });
-    }
-
     publishMessage(message, { parentMessageId, threadId } = {}, headers = {}) {
+        let msgHeaders = Object.assign({
+            execId: env.ELASTICIO_EXEC_ID,
+            taskId: env.ELASTICIO_FLOW_ID,
+            workspaceId: env.ELASTICIO_WORKSPACE_ID,
+            userId: env.ELASTICIO_USER_ID,
+            threadId,
+            stepId: message.headers.stepId,
+            messageId: parentMessageId
+        }, headers);
+        const protocolVersion = Number(msgHeaders.protocolVersion || 1);
         return this.subscriptionChannel.publish(
             env.ELASTICIO_LISTEN_MESSAGES_ON,
             env.ELASTICIO_DATA_ROUTING_KEY,
-            new Buffer(JSON.stringify(message)), {
-                headers: Object.assign({
-                    execId: env.ELASTICIO_EXEC_ID,
-                    taskId: env.ELASTICIO_FLOW_ID,
-                    workspaceId: env.ELASTICIO_WORKSPACE_ID,
-                    userId: env.ELASTICIO_USER_ID,
-                    threadId,
-                    stepId: message.headers.stepId,
-                    messageId: parentMessageId
-                }, headers)
-            });
+            encryptor.encryptMessageContent(
+                message,
+                protocolVersion < 2 ? 'base64' : undefined
```

Comment:
why to not put handling protocol version inside `encryptMessageContent `

Reply:
@VitaliyKovalchuk cause we do not always have full message with determined headers to encrypt/decrypt (see errors logic)